src/LanguageServer/Protocol/Handler/Extensions/ExtensionDocumentMessageHandler.cs

```diff
@@ -31,9 +31,13 @@ public async Task<ExtensionMessageResponse> HandleRequestAsync(ExtensionDocument
         var solution = context.Document.Project.Solution;
 
         var service = solution.Services.GetRequiredService<IExtensionMessageHandlerService>();
-        var response = await service.HandleExtensionDocumentMessageAsync(
+        var (response, exception) = await service.HandleExtensionDocumentMessageAsync(
             context.Document, request.MessageName, request.Message, cancellationToken).ConfigureAwait(false);
 
+        // Report any exceptions the extension itself caused while handling the request.
+        if (exception is not null)
+            context.Logger.LogException(exception);
+
         return new ExtensionMessageResponse(response);
```

Comment:
This cannot be left as-is: the extension should be notified of the exception, otherwise they have no way to diagnose and fix the problem. I understand that you will fix this before completing the PR. But I am leaving this comment to clarify that we are missing something :-)

Reply:
yes.  followup will update the LSP protocol between you and us to pass this data back as well.